app/main.dev.js

```diff
@@ -163,16 +165,44 @@ app.on('ready', async () => {
   }
 
   /**
-   * Add application event listener:
-   *  - Open zap payment form when lightning url is opened
+   * Handler for lightning: links
    */
-  app.setAsDefaultProtocolClient('lightning')
-  app.on('open-url', (event, url) => {
-    mainLog.debug('open-url')
-    event.preventDefault()
-    const payReq = url.split(':')[1]
+  const handleLightningLink = payReq => {
     zap.sendMessage('lightningPaymentUri', { payReq })
     zap.mainWindow.show()
+  }
+
+  /**
+   * Handler for lndconnect: links
+   */
+  const handleLndconnectLink = query => {
+    const { host, cert, macaroon } = querystring.parse(query)
+    zap.sendMessage('lndconnectUri', { host, cert, macaroon })
+    zap.mainWindow.show()
+  }
+
+  /**
+   * Add application event listeners:
+   *  - lightning: Open zap payment form when lightning url is opened
+   *  - lndconnect: Populate onboarding connection details form when lndconnect url is opened
+   */
+  app.setAsDefaultProtocolClient('lightning')
+  app.setAsDefaultProtocolClient('lndconnect')
+
+  app.on('open-url', (event, input) => {
+    mainLog.debug('open-url: %s', input)
+    event.preventDefault()
+
+    const parsedUrl = url.parse(input)
+
+    switch (parsedUrl.protocol) {
```

Comment:
should we add a `default` to this switch case and throw an error to the console? I envision this being helpful when other services build in support to connect to zap and they aren't inputting the correct uri handler, this way debugging issues will be easier.

Reply:
We only handle `lightning` and `lndconnect` urls, so these are the only 2 things that can possibly trigger this part of the code.

When we call `sendMessage` that is already logged to the console, so it's quite easy to see both the input and output of these handlers (we log the input above).

Reply:
ah yes of course you have to register the handlers to fire those callbacks anyways 👍